### PR TITLE
Fix coins codegen

### PR DESCRIPTION
**Pull request by leoneparise**
## Description
The correct method to **filter** values in ruby is `select`. Filter is not available on most environments. This PR fixes this issue.

## Testing instructions

Just run `./bootstrap`. I should compile without errors.

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.

### CODE REVIEW

**Comment by hewigovens** [comment]
from the docs `filter` is an alias of `select`, our CI is already using ruby 2.6, so I don't think it's a problem now

File: codegen/bin/coins [hunk 1]
```diff
@@ -38,5 +38,5 @@ erbs.each do |erb|
   result = template.result(binding)
 
   path = File.join(erb['folder'], erb['file'])
-  File.write(path, result)
+  File.write(path, result)
```
Comment (wpank) [inline codegen/bin/coins]: We can add a git pre-commit or pre-push hook that automatically formats/removes whitespace using something like [CLangFormat](https://clang.llvm.org/docs/ClangFormat.html) for things like this. Not entirely necessary, but perhaps something to consider.

Comment (leoneparise) [reply codegen/bin/coins]: Agreed. Sometimes the editor add those spaces and I only notice when I push the branch.

**Comment by leoneparise** [comment]
> from the docs `filter` is an alias of `select`, our CI is already using ruby 2.6, so I don't think it's a problem now

MacOS default ruby is 2.3.7. Most developers will use this version, if filter is only an `alias` to `select`, why not use `select`, which is supported on most environments?

**Pull request merged**